src/main/java/org/amahi/anywhere/util/Downloader.java

```diff
@@ -66,6 +67,12 @@ private void setUpDownloadReceiver() {
     }
 
     private void startDownloading(Uri downloadUri, String downloadName) {
+        
+        ///code to delete the file if it already exists
+        File file = new File(Environment.getExternalStorageDirectory()+"/"+Environment.DIRECTORY_DOWNLOADS+"/"+downloadName);
```

Comment:
You are checking for the file in the wrong directory. The file is saved in `Android/data/org.amahi.anywhere/files/Download/filename.ext` where as you are looking in `Downloads/filename.ext`. Check the docs here: [Environment.getExternalStorageDirectory()](https://developer.android.com/reference/android/os/Environment.html#getExternalStorageDirectory()) and [Context.getExternalFilesDir(String)](https://developer.android.com/reference/android/content/Context.html#getExternalFilesDir(java.lang.String)).

So replacing `Environment.getExternalStorageDirectory()+"/"+Environment.DIRECTORY_DOWNLOADS` with `context.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS)` will do the job!